src/task.c

```diff
@@ -281,8 +281,9 @@ task_finish_callback (gint pid_result, gboolean localwatchdog, gpointer user_dat
     if (pid_result == 0) {
         task->state = task_run_data->pass_state;
 
-        // If not running in rhts compat mode report PASS for exit code 0
-        if (!task->rhts_compat) {
+        // If not running in rhts compat mode and user hasn't reported results,
+        // report PASS for exit code 0.
+        if (!task->rhts_compat && !task->results_reported) {
```

Comment:
How about restricting this to not rhts_compat and not reported skip task?  It would continue to report pass or fail if the task hasn't been skipped.

Reply:
Results reported by rstrnt-report-result takes precedence.  I felt checking whether it was called by user should suffice and not corner case the check to SKIP.  Why initiate PASS again when user has already reported results?

Reply:
So this will still report fail if the script exits non-zero.  I looked too quickly before.  When you said second PASS in your comment I looked closer.  I apologize for not looking closer the first time. :-)  Thanks!

Reply:
For non-zero, I gave it a lot of thought and felt it was best to leave as is.  It gives the user the option to continue
their job(FAIL) or they can call rstrnt-abort and simulate legacy code which set task status to ABORT.